ee/agent/internal/informer_stacks.go

```diff
@@ -48,28 +56,43 @@ func StacksEventHandler(logger sharedlogging.Logger, membershipClient Membership
 		}())
 	}
 
-	return cache.ResourceEventHandlerFuncs{
-		AddFunc: func(obj interface{}) {
-			stack := convertUnstructured[*v1beta1.Stack](obj)
-			logger.Infof("Stack '%s' added", stack.Name)
-			sendStatusFromStack(stack)
-		},
-		UpdateFunc: func(oldObj, newObj interface{}) {
+	return &StackEventHandler{
+		logger: logger,
+		ResourceEventHandlerFuncs: cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
 
-			oldStack := convertUnstructured[*v1beta1.Stack](oldObj)
-			newStack := convertUnstructured[*v1beta1.Stack](newObj)
+				stack := convertUnstructured[*v1beta1.Stack](obj)
+				if _, ok := stacks[stack.Name]; !ok {
+					logger.Debugf("Stack '%s' not initialized in memory", stack.Name)
+					return
+				}
 
-			if oldStack.Spec.Disabled == newStack.Spec.Disabled && oldStack.Status.Ready == newStack.Status.Ready {
-				return
-			}
+				logger.Infof("Stack '%s' added", stack.Name)
+				sendStatusFromStack(stack)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
 
-			logger.Infof("Stack '%s' updated", newStack.Name)
-			sendStatusFromStack(newStack)
-		},
-		DeleteFunc: func(obj interface{}) {
-			stack := convertUnstructured[*v1beta1.Stack](obj)
-			logger.Infof("Stack '%s' deleted", stack.Name)
-			sendStatus(stack.Name, generated.StackStatus_Deleted)
+				newStack := convertUnstructured[*v1beta1.Stack](newObj)
+
+				if _, ok := stacks[newStack.Name]; !ok {
+					logger.Debugf("Stack '%s' not initialized in memory", newStack.Name)
```

Comment:
I'm not sure it will be ok to just ignore the event

Reply:
It's a side effect, where fisrt,
- i was having a stack in progressing while i was ignoring message when !inmemory OR (old,new) was equal
- I'll check again, but this send a status each informer loop,

Reply:
That's why i've deleted this check, now it' ok both statuses are set correctly and tested